src/lib/core.d.ts

```diff
@@ -1011,12 +1011,12 @@ interface Array<T> {
       * Combines two or more arrays.
       * @param items Additional items to add to the end of array1.
       */
-    concat<U extends T[]>(...items: U[]): T[];
+    concat<U extends T[]>(...items: (U | U[])[]): T[];

```

Comment:
The generic overload should go. It was there for historical reasons (it was a workaround for an issue in the old compiler having to do with instantiations of `Array<T>` within `Array<T>` itself).